Add first question to survey

katrinacrisostomo opened the pull request:
Issue: https://github.com/uwblueprint/somo/issues/45

Please don't merge until Bree and Nick take a look as I wasn't able to finish it with them

katrinacrisostomo committed:
Add first question ID field to surveys table

katrinacrisostomo committed:
merge conflict

katrinacrisostomo committed:
Add survey and first_question association to survey and question models

katrinacrisostomo committed:
Add first question validation spec to survey

chrisjluc commented on app/models/survey.rb:
```diff
@@ -7,11 +7,14 @@
 #  description                          :text
 #  created_at                           :datetime         not null
 #  updated_at                           :datetime         not null
+#  first_question_id                    :integer
```
Nit: Can we move this up maybe 2nd or 3rd from the top, because it's more important than `created_at` and `updated_at`

katrinacrisostomo replied:
👍 good point

chrisjluc commented on app/models/survey.rb:
```diff
@@ -7,11 +7,14 @@
-  validates :description, :name, presence: true
+  belongs_to :first_question, foreign_key: :first_question_id, class_name: 'Question'
+
+  validates :description, :name, :first_question_id, presence: true
```
I don't think a `first_question_id` needs to be present for a valid survey.

Example scenarios:
- User creates survey, doesn't add first question
- User creates survey with survey title and description, but doesn't add first question

They should be able to save their surveys in the above scenarios.

katrinacrisostomo replied:
okaaay true, I was thinking we wouldn't allow them to create a survey without any questions but I guess why not

chrisjluc commented on spec/models/survey_spec.rb:
```diff
@@ -10,4 +10,9 @@
     it { is_expected.not_to be_valid }
   end
+
+  context "survey is missing a first question" do
```
Need to change this test, if we all first_question to be empty

katrinacrisostomo replied:
I guess I would just delete this test altogether!

katrinacrisostomo committed:
Merge branch 'master' of https://github.com/uwblueprint/somo into add_first_question_to_survey

katrinacrisostomo committed:
Address code review comments

Move first_question_id up in the schema comments
Remove validation of first_question_id in survey model
Remove corresponding first_question_id validation spec in survey specs

katrinacrisostomo commented:
Addressed comments, ready for another pass of code review 👍

chrisjluc commented on spec/factories/surveys.rb:
```diff
@@ -2,5 +2,6 @@
   factory :survey do
     name 'hello world'
     description 'a test survey'
+    first_question_id 1
```
Why is this set to 1?

I think it's better to replace this line with `first_question` with nothing after because I think the factory will create a new real record automagically. While, what you have it will set it to 1, but there might not be a question with id 1.

katrinacrisostomo replied:
oh ya my bad, forgot to delete with the spec

katrinacrisostomo committed:
delete first_question_id in survey factory

chrisjluc commented:
Hmmm might be better to have `first_question` by default, because that seems like the more common test case rather than survey without first question, instead of just deleting it.

katrinacrisostomo committed:
Add first question id back

katrinacrisostomo commented:
Okaaay done

katrinacrisostomo committed:
delete line again

katrinacrisostomo committed:
Fix factories

chrisjluc commented:
Can you write a test to check that is valid and that it's a question model?

Similar to `question_spec.rb`

katrinacrisostomo committed:
Fix stuffs hehehehehhe

chrisjluc commented:
LGTM
